---
float32[6] tempvalue